tests/CoordinateServiceTest.cs

```diff
@@ -0,0 +1,197 @@
+        Assert.AreEqual(6, CoordinateService.GetDecimalPlaces(100000.123456));
+        Assert.AreEqual(2, CoordinateService.GetDecimalPlaces(10000000.01));
+        Assert.AreEqual(0, CoordinateService.GetDecimalPlaces(1000000));
+    }
+
+    [TestMethod]
+    public async Task MigrateCoordinatesOfLV95BoreholeWithAllCoordinatesSet()
+    {
+        httpClientFactoryMock.Setup(cf => cf.CreateClient(It.IsAny<string>())).Returns(new HttpClient()).Verifiable();
+
+        var borehole = CreateLV95BoreholeWithAllCoordinatesSet();
+
+        await service.MigrateCoordinatesOfBorehole(borehole, false);
+        context.SaveChanges();
+
+        var result = context.Boreholes.FirstOrDefault(b => b.Id == LV95BoreholeWithAllCoordinatesSetId);
+        Assert.AreEqual(ReferenceSystem.LV95, result.OriginalReferenceSystem);
+        Assert.AreEqual(2626103.1988343936, result.LocationX);
+        Assert.AreEqual(1125366.3469565178, result.LocationY);
+        Assert.AreEqual(626103.56923180178, result.LocationXLV03);
+        Assert.AreEqual(125366.57802526229, result.LocationYLV03);
+    }
+
+    [TestMethod]
+    public async Task NotMigrateCoordinatesOfLV95BoreholeWithAllCoordinatesSet()
+    {
+        httpClientFactoryMock.Setup(cf => cf.CreateClient(It.IsAny<string>())).Returns(new HttpClient()).Verifiable();
+
+        var borehole = CreateLV95BoreholeWithAllCoordinatesSet();
+
+        // Only calculate missing values.
+        await service.MigrateCoordinatesOfBorehole(borehole, true);
+        context.SaveChanges();
+
+        var result = context.Boreholes.FirstOrDefault(b => b.Id == LV95BoreholeWithAllCoordinatesSetId);
+        Assert.AreEqual(ReferenceSystem.LV95, result.OriginalReferenceSystem);
+        Assert.AreEqual(2626103.1988343936, result.LocationX);
+        Assert.AreEqual(1125366.3469565178, result.LocationY);
+        Assert.AreEqual(765875.1615463407, result.LocationXLV03);
+        Assert.AreEqual(78390.10392298926, result.LocationYLV03);
+    }
+
+    [TestMethod]
+    public async Task MigrateCoordinatesOfLV03BoreholeWithMissingDestCoordinates()
+    {
+        httpClientFactoryMock.Setup(cf => cf.CreateClient(It.IsAny<string>())).Returns(new HttpClient()).Verifiable();
+
+        var borehole = CreateLV03BoreholeWithhMissingDestCoordinates();
+
+        await service.MigrateCoordinatesOfBorehole(borehole, false);
+        context.SaveChanges();
+
+        var result = context.Boreholes.FirstOrDefault(b => b.Id == LV03BoreholeWithMissingDestCoordinatesId);
+        Assert.AreEqual(ReferenceSystem.LV03, result.OriginalReferenceSystem);
+        Assert.AreEqual(2655270, result.LocationX);
+        Assert.AreEqual(1297874, result.LocationY);
+        Assert.AreEqual(655269, result.LocationXLV03);
+        Assert.AreEqual(297874, result.LocationYLV03);
+    }
+
+    [TestMethod]
+    public async Task NotMigrateCoordinatesOfLV03BoreholeWithMissingSourceCoordinates()
```

Comment:
PP: Der Methodenname klingt ein wenig holprig. Alternativvorschlag:
```suggestion
    public async Task DoesNotMigrateCoordinatesOfLV03BoreholeWithMissingSourceCoordinates()
```